lib/GridItem.jsx

```diff
@@ -161,6 +169,42 @@ export default class GridItem extends React.Component<Props, State> {
     className: ""
   };
 
+  currentNode: HTMLElement;
+
+  componentWillReceiveProps(nextProps: Props) {
```

Comment:
As you might be aware, `componentWillReceiveProps`  will be unsupported as of React version 17. [Here](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops) is more info. We should try to stay away from it, and in fact re-write the logic in other places of the project.

Reply:
@n1ghtmare I have a solution in my fork. I think it'll be better to replace all componentWillReceiveProps methods in a separate PR